src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -595,13 +653,257 @@ public static UntypedResultSet loadCommandRow(CommandStore commandStore, TxnId t
+                    callback.onNext(row);
+                    lastRow = row;
+                }
+                if (shouldContinue(lastRow))
+                {
+                    lastSeen = lastRow;
+                    schedule();
+                }
+                else
+                {
+                    lastSeen = null;
+                }
+            }
+            catch (Throwable t)
+            {
+                callback.onError(t);
+            }
+        }
+
+        @Override
+        public long creationTimeNanos()
+        {
+            return creationTimeNanos;
+        }
+
+        @Override
+        public long startTimeNanos()
+        {
+            return startTimeNanos;
+        }
+
+        @Override
+        public String description()
+        {
+            return String.format("Table Walker for %s; queries = %d", getClass().getSimpleName(), numQueries);
+        }
+    }
+
+    private static String selection(TableMetadata metadata, Set<String> requiredColumns, Set<String> forIteration)
+    {
+        StringBuilder selection = new StringBuilder();
+        if (requiredColumns.isEmpty())
+            selection.append("*");
+        else
+        {
+            Sets.SetView<String> other = Sets.difference(requiredColumns, forIteration);
+            for (String name : other)
+            {
+                ColumnMetadata meta = metadata.getColumn(new ColumnIdentifier(name, true));
+                if (meta == null)
+                    throw new IllegalArgumentException("Unknown column: " + name);
+            }
+            List<String> names = new ArrayList<>(forIteration.size() + other.size());
+            names.addAll(forIteration);
+            names.addAll(other);
+            // this sort is to make sure the CQL is determanistic
+            Collections.sort(names);
+            for (int i = 0; i < names.size(); i++)
+            {
+                if (i > 0)
+                    selection.append(", ");
+                selection.append(names.get(i));
+            }
+        }
+        return selection.toString();
+    }
+
+    private static class WalkCommandsForDomain extends TableWalk
+    {
+        private static final Set<String> COLUMNS_FOR_ITERATION = ImmutableSet.of("txn_id", "store_id", "domain");
+        private final String cql;
+        private final int storeId, domain;
+
+        private WalkCommandsForDomain(int commandStore, Routable.Domain domain, Set<String> requiredColumns, Executor executor, Observable<UntypedResultSet.Row> callback)
+        {
+            super(executor, callback);
+            this.storeId = commandStore;
+            this.domain = domain.ordinal();
+            cql = String.format("SELECT %s " +
+                                "FROM %s " +
+                                "WHERE store_id = ? " +
+                                "      AND domain = ? " +
+                                "      AND token(store_id, domain, txn_id) > token(?, ?, (?, ?, ?)) " +
```

Comment:
since we know this table has a token layout that matches the data (local partitioner) this helps make sure that the exec engine does what we want.